File: app/src/test/java/com/soundcloud/android/image/ImageCacheTest.java
package com.soundcloud.android.image;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;

import com.nostra13.universalimageloader.cache.disc.naming.FileNameGenerator;
import com.soundcloud.android.model.Urn;
import com.soundcloud.android.properties.ApplicationProperties;
import com.soundcloud.android.testsupport.AndroidUnitTest;
import com.soundcloud.android.utils.DeviceHelper;
import com.soundcloud.android.utils.cache.Cache;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;

import android.graphics.Bitmap;
import android.graphics.drawable.TransitionDrawable;

public class ImageCacheTest extends AndroidUnitTest {

    private static final String RESOLVER_URL = "https://api-mobile.soundcloud.com/images/soundcloud:tracks:1/large";

    private ImageCache imageCache;

    private Cache<String, TransitionDrawable> placeholderCache = Cache.withSoftValues(10);
    private Cache<Urn, Bitmap> blurredImageCache = Cache.withSoftValues(10);

    @Mock TransitionDrawable cachedTransitionDrawable;
    @Mock TransitionDrawable generatedTransitionDrawable;

    @Mock FileNameGenerator fileNameGenerator;
    @Mock UserAgentImageDownloaderFactory imageDownloaderFactory;
    @Mock DeviceHelper deviceHelper;
    @Mock ApplicationProperties properties;

    @Before
    public void setUp() throws Exception {
        imageCache = new ImageCache(placeholderCache, blurredImageCache, context(), properties, fileNameGenerator, imageDownloaderFactory, deviceHelper);
    }

    @Test
    public void getPlaceholderDrawableReturnsDrawable() throws Exception {
        placeholderCache.put(RESOLVER_URL + "_100_100", cachedTransitionDrawable);

        TransitionDrawable placeholderDrawable = imageCache.getPlaceholderDrawable(RESOLVER_URL, 100, 100, mock(PlaceholderGenerator.class));

        assertThat(placeholderDrawable).isSameAs(cachedTransitionDrawable);
    }

}

File: app/src/test/java/com/soundcloud/android/image/ImageCacheTest.kt
package com.soundcloud.android.image

import android.graphics.Bitmap
import android.graphics.drawable.TransitionDrawable
import com.soundcloud.android.model.Urn
import com.soundcloud.android.testsupport.AndroidUnitTest
import com.soundcloud.android.utils.cache.Cache
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.mockito.Mock

class ImageCacheTest : AndroidUnitTest() {

    private lateinit var imageCache: ImageCache

    private val placeholderCache = Cache.withSoftValues<String, TransitionDrawable>(10)
    private val blurredImageCache = Cache.withSoftValues<Urn, Bitmap>(10)

    @Mock internal lateinit var cachedTransitionDrawable: TransitionDrawable
    @Mock internal lateinit var placeHolderGenerator: PlaceholderGenerator

    @Before
    @Throws(Exception::class)
    fun setUp() {
        imageCache = ImageCache(placeholderCache, blurredImageCache)
    }

    @Test
    @Throws(Exception::class)
    fun getPlaceholderDrawableReturnsDrawable() {
        placeholderCache.put("https://api-mobile.soundcloud.com/images/soundcloud:tracks:1/large_100_100", cachedTransitionDrawable)

        val placeholderDrawable = imageCache.getPlaceholderDrawable("https://api-mobile.soundcloud.com/images/soundcloud:tracks:1/large", 100, 100, placeHolderGenerator)

        assertThat(placeholderDrawable).isSameAs(cachedTransitionDrawable)
    }

}